go-controller/pkg/libovsdb/ops/portgroup.go

```diff
@@ -27,9 +28,10 @@ func CreateOrUpdatePortGroupsOps(nbClient libovsdbclient.Client, ops []libovsdb.
 	opModels := make([]operationModel, 0, len(pgs))
 	for i := range pgs {
 		pg := pgs[i]
+		updatedFields := getFieldsToUpdate(pg, getDefaultUpdatableFields(pg))
```

Comment:
missed this:  why indirect `getDefaultUpdatableFields` through `getFieldsToUpdate`? We can just have the old implementation of `getAllUpdatableFields`

Reply:
now `getDefaultUpdatableFields` returns the `[]ModelUpdateField` list (and not []interface{}), and the only function converting them to []interface{} is `getFieldsToUpdate`. I wanted only one function to do the conversion..

Reply:
There is no reason to do this conversion at all.

Just have `getDefaultUpdatableFields` return `[]interface{}` and `createOrUpdateLogicalSwitchPortsOps` take `[]interface{}`. Use `ModelUpdateField`only where strictly needed.

Reply:
Ok, my bad, I see that `createOrUpdateLogicalSwitchPortsOps` also takes a list of ports. It would be best if it relied on a method handling one port at a time. Let me know what you want to do.

Reply:
It may be best to keep both functions then (`getAllUpdatableFields` that returns the []interface{} and `getDefaultUpdatableFields` that returns the new enum list), for more flexibility, wdyt?

Reply:
>It would be best if it relied on a method handling one port at a time. Let me know what you want to do.

The function gets a list of LSPs, and does the same "updates" for each LSP. running `getDefaultUpdatableFields` for each LSP seems redundant. 

I see the point in every suggestion, but nothing fits all cases.. This is why I suggest the above solution of two functions that do similar things..

Reply:
I don't have anything against the two functions other than they would have to be kept in sync.

> running `getDefaultUpdatableFields` for each LSP seems redundant.

That's unavoidable. At the end of the day we have to pass down a specific pointer for each field of each LSP.

Reply:
> I don't have anything against the two functions other than they would have to be kept in sync.
> 
> > running `getDefaultUpdatableFields` for each LSP seems redundant.
> 
> That's unavoidable. At the end of the day we have to pass down a specific pointer for each field of each LSP.

good point :) 

> 
> Just laying out everything on the table so that we make an informed decision. At this point I don't care that much about the two options. If I came across modifying that code however I would probably try to habe `createOrUpdateLogicalSwitchPortsOps` call a `createOrUpdateLogicalSwitchPortOps` that would just handle one port.

That's not a problem to do. tell me what you think ([here](https://github.com/ovn-kubernetes/ovn-kubernetes/compare/c790b3614065d288f4bd0c4f9043af133ed6c122..479446d0c3777bd501693cd6cc4264080f2f73b1))